### PR TITLE
Improve checks for latex support

### PR DESCRIPTION
**Pull request by andreas-bok-sociomantic**
It is quite frustrating build the `pdf` manual for **libdrizzle-redux** if not all the required `latex` packages are installed. 

This PR has the following parts 
- adds `autoconf` macros which checks for the programs, `latex`, `pdflatex` and `kpsewhich` 
- adds a **bash** script which tests whether a required package is available using `kpsewhich`

### CODE REVIEW

File: m4/acltx_prog_kpsewhich.m4 [hunk 1]
```diff
@@ -0,0 +1,20 @@
+dnl @synopsis ACLTX_PROG_KPSEWHICH([ACTION-IF-NOT-FOUND])
+dnl
+dnl This macro find a kpsewhich application and set the variable
+dnl kpsewhich to the name of the application or to no if not found if
+dnl ACTION-IF-NOT-FOUND is not specified, configure fail when then
+dnl application is not found.
```
Comment (ben-palmer-sociomantic) [inline m4/acltx_prog_kpsewhich.m4]: This whole paragraph is very strangely worded. Perhaps it should be something more like:

```
This macro looks for the kpsewhich application and sets the variable
kpsewhich to the name of the application if found or no if not found. If
ACTION-IF-NOT-FOUND is not specified when application is not found then fail.
```

File: m4/acltx_prog_latex.m4 [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# SYNOPSIS
+#
+#   ACLTX_PROG_LATEX([ACTION-IF-NOT-FOUND])
+#
+# DESCRIPTION
+#
+#   This macro find a latex application and set the variable latex to
+#   the name of the application or to no if not found if
+#   ACTION-IF-NOT-FOUND is not specified, configure fail when then
+#   application is not found.
```
Comment (ben-palmer-sociomantic) [inline m4/acltx_prog_latex.m4]: Again this paragraph is strangely worded.

File: m4/acltx_prog_latex.m4 [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#
+#   It is possible to set manually the program to use using latex=...
```
Comment (ben-palmer-sociomantic) [inline m4/acltx_prog_latex.m4]: should be `is possible to manually set`

File: m4/acltx_prog_pdflatex.m4 [hunk 4]
```diff
@@ -0,0 +1,20 @@
+dnl @synopsis AC_PROG_PDFLATEX
+dnl
+dnl This macro find a pdflatex application and set the variable
+dnl pdflatex to the name of the application or to no if not found if
+dnl ACTION-IF-NOT-FOUND is not specified, configure fail when then
+dnl application is not found.
```
Comment (ben-palmer-sociomantic) [inline m4/acltx_prog_pdflatex.m4]: Again this paragraph needs rewording.

**Comment by ben-palmer-sociomantic** [comment]
The wording of the commit title and message for the commit `Use variable for description` are a bit unclear. The title could be more clear so maybe `Use variable for documentation description` or something like it would be more specific. The message is not grammatically correct, perhaps it should be `Store description as a variable instead of as a string literal`.

**Comment by andreas-bok-sociomantic** [comment]
> Again this paragraph is strangely worded.

My bad, didn't actually read through the autoconf macro files, since they are added directly from the autoconf repo, http://ac-archive.sourceforge.net/ac-archive/

I will change review them, and also add a PR against the autoconf repo.

**Comment by andreas-bok-sociomantic** [comment]
Reworded Yoda speech 

```diff
diff --git a/m4/acltx_prog_kpsewhich.m4 b/m4/acltx_prog_kpsewhich.m4
index c73b4b6..fc68295 100644
--- a/m4/acltx_prog_kpsewhich.m4
+++ b/m4/acltx_prog_kpsewhich.m4
@@ -1,9 +1,10 @@
 dnl @synopsis ACLTX_PROG_KPSEWHICH([ACTION-IF-NOT-FOUND])
 dnl
-dnl This macro find a kpsewhich application and set the variable
-dnl kpsewhich to the name of the application or to no if not found if
-dnl ACTION-IF-NOT-FOUND is not specified, configure fail when then
-dnl application is not found.
+dnl Tests if the application kpsewhich exists. If so, the variable $kpsewhich is
+dnl assigned the name of the application, or 'no' if kpsewhich was not found.
+dnl    In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+dnl    specified
+dnl
 dnl
 dnl @category LaTeX
 dnl @author Boretti Mathieu <boretti@eig.unige.ch>
diff --git a/m4/acltx_prog_latex.m4 b/m4/acltx_prog_latex.m4
index 9563740..8e998e7 100644
--- a/m4/acltx_prog_latex.m4
+++ b/m4/acltx_prog_latex.m4
@@ -6,12 +6,12 @@
 #
 # DESCRIPTION
 #
-#   This macro find a latex application and set the variable latex to
-#   the name of the application or to no if not found if
-#   ACTION-IF-NOT-FOUND is not specified, configure fail when then
-#   application is not found.
+#   Check if a latex application exists and set the variable $latex
+#   to the name of the application or to 'no' if none was found.
+#      In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+#      specified
 #
-#   It is possible to set manually the program to use using latex=...
+#   It is possible to manually set the program using latex=...
 #
 # LAST MODIFICATION
 #
diff --git a/m4/acltx_prog_pdflatex.m4 b/m4/acltx_prog_pdflatex.m4
index 7781e59..6d543c8 100644
--- a/m4/acltx_prog_pdflatex.m4
+++ b/m4/acltx_prog_pdflatex.m4
@@ -1,9 +1,9 @@
-dnl @synopsis AC_PROG_PDFLATEX
+dnl @synopsis ACLTX_PROG_PDFLATEX([ACTION-IF-NOT-FOUND])
 dnl
-dnl This macro find a pdflatex application and set the variable
-dnl pdflatex to the name of the application or to no if not found if
-dnl ACTION-IF-NOT-FOUND is not specified, configure fail when then
-dnl application is not found.
+dnl Tests if the pdflatex application exists and sets the variable $pdflatex to
+dnl the name of the application or to 'no', if it was not found.
+dnl    In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+dnl    specified
 dnl
 dnl @category LaTeX
 dnl @author Boretti Mathieu <boretti@eig.unige.ch>

```

File: docs/conf.py.in [hunk 5]
```diff
@@ -188,7 +188,8 @@ htmlhelp_basename = 'gearmandoc'
 # Grouping the document tree into LaTeX files. List of tuples
 # (source start file, target name, title, author, documentclass [howto/manual]).
 latex_documents = [
-  ('index', 'libdrizzle.tex', description, u'Drizzle Development Group', 'manual'),
+  ('index', 'libdrizzle.tex', description,
+   u'Drizzle Development Group', 'manual'),
```
Comment (ben-palmer-sociomantic) [inline docs/conf.py.in]: I don't really get what this change is doing based on the commit message.

File: m4/acltx_prog_kpsewhich.m4 (hunk 1 above)
Comment (ben-palmer-sociomantic) [reply m4/acltx_prog_kpsewhich.m4]: I still thing this paragraph is strangely worded.

File: m4/acltx_prog_latex.m4 (hunk 2 above)
Comment (ben-palmer-sociomantic) [reply m4/acltx_prog_latex.m4]: Same here

File: m4/acltx_prog_pdflatex.m4 (hunk 4 above)
Comment (ben-palmer-sociomantic) [reply m4/acltx_prog_pdflatex.m4]: Same here.

File: docs/latex_check_packages.sh [hunk 6]
```diff
@@ -0,0 +1,97 @@
+##
+# shell script which checks whether required packages
+# is available for building the pdf documentation
```
Comment (ben-palmer-sociomantic) [inline docs/latex_check_packages.sh]: Should be `required packages are available`.

File: m4/acltx_prog_kpsewhich.m4 [hunk 7]
```diff
@@ -1,9 +1,10 @@
 dnl @synopsis ACLTX_PROG_KPSEWHICH([ACTION-IF-NOT-FOUND])
 dnl
-dnl This macro find a kpsewhich application and set the variable
-dnl kpsewhich to the name of the application or to no if not found if
-dnl ACTION-IF-NOT-FOUND is not specified, configure fail when then
-dnl application is not found.
+dnl Tests if the application kpsewhich exists. If so, the variable $kpsewhich is
```
Comment (ben-palmer-sociomantic) [inline m4/acltx_prog_kpsewhich.m4:3]: Ah I didn't see this commit when I made my other comments. You can ignore them.

**Review by ben-palmer-sociomantic — COMMENTED** [review]
I made a couple of comments bu hadn't seen the last commit so some of them can be ignored.

**Comment by andreas-bok-sociomantic** [comment]
Addressed comments and squashed commits
```diff
 # The name of an image file (relative to this directory) to place at the top of
diff --git a/docs/latex_check_packages.sh b/docs/latex_check_packages.sh
index a4d62b7..a6691d3 100755
--- a/docs/latex_check_packages.sh
+++ b/docs/latex_check_packages.sh
@@ -1,6 +1,6 @@
 ##
 # shell script which checks whether required packages
-# is available for building the pdf documentation
+# are available for building the pdf documentation
 #
```

**Comment by andreas-bok-sociomantic** [comment]
The changes on `docs/conf.py.in` reverted an earlier commit removing the file completely from this PR

File: m4/acltx_prog_kpsewhich.m4 [hunk 8]
```diff
@@ -0,0 +1,21 @@
+dnl @synopsis ACLTX_PROG_KPSEWHICH([ACTION-IF-NOT-FOUND])
+dnl
+dnl Tests if the application kpsewhich exists. If so, the variable $kpsewhich is
+dnl assigned the name of the application, or 'no' if kpsewhich was not found.
+dnl	In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+dnl	specified
```
Comment (ben-palmer-sociomantic) [inline m4/acltx_prog_kpsewhich.m4]: Is there some weird indentation here?

File: m4/acltx_prog_latex.m4 [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+# SYNOPSIS
+#
+#   ACLTX_PROG_LATEX([ACTION-IF-NOT-FOUND])
+#
+# DESCRIPTION
+#
+#   Check if a latex application exists and set the variable $latex
+#   to the name of the application or to 'no' if none was found.
+#	In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+#	specified
```
Comment (ben-palmer-sociomantic) [inline m4/acltx_prog_latex.m4]: Is there some weird indentation here?

File: m4/acltx_prog_pdflatex.m4 [hunk 10]
```diff
@@ -0,0 +1,20 @@
+dnl @synopsis ACLTX_PROG_PDFLATEX([ACTION-IF-NOT-FOUND])
+dnl
+dnl Tests if the pdflatex application exists and sets the variable $pdflatex to
+dnl the name of the application or to 'no', if it was not found.
+dnl	In the latter case, configure will fail, if ACTION-IF-NOT-FOUND is not
+dnl	specified
```
Comment (ben-palmer-sociomantic) [inline m4/acltx_prog_pdflatex.m4]: Is there some weird indentation here?

File: docs/latex_check_packages.sh [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+printf "\n%-22s Installed\n" "Package"
+printf "================================\n"
+# iterate required packages
+for i in $(echo "$LATEX_PACKAGES");
+do
+  f="$i.sty"
+  printf "%-22s " $f;
+  FOUND="Yes"
+  # adds package to MISSING packages if the can't be found
+  # by  neither kpsewhich nor find
```
Comment (ben-palmer-sociomantic) [inline docs/latex_check_packages.sh]: extra space between `by` and `neither`

**Review by ben-palmer-sociomantic — COMMENTED** [review]
Couple of minor comments on documentation, otherwise LGTM.

**Comment by andreas-bok-sociomantic** [comment]
Fixed weird indentation. 
The build fails on travis due to an error caused by the mysql service being started.

**Comment by andreas-bok-sociomantic** [comment]
The build error could be related to this change at Travis https://docs.travis-ci.com/user/build-environment-updates/2016-12-01/#Details, with reported issues https://github.com/travis-ci/travis-ci/issues/6961

**Comment by andreas-bok-sociomantic** [comment]
Updated
- fixed a few errors, 
- rebased on newest v5.x.x branch

**Comment by ben-palmer-sociomantic** [comment]
LGTM

**Pull request merged**